code_annotations/django_reporting_helpers.py

```diff
@@ -0,0 +1,90 @@
+"""
+Django related functionality for code annotations.
+"""
+import inspect
+import sys
+
+from django.apps import apps
+from django.db import models
+
+
+def requires_pii_annotations(model):
+    """
+    Return true if the given model actually requires annotations, according to PLAT-2344.
+    """
+    return issubclass(model, models.Model) \
+        and not (model is models.Model) \
+        and not model._meta.abstract \
+        and not model._meta.proxy
+
+
+def is_non_local(model):
+    """
+    Determine if the given model is non-local to the current IDA.
+
+    Non-local models are all installed models which are not "local", by
+    definition.  "Local" models are installed models which are defined in code
+    which physically lives in the current codebase, where "current codebase"
+    ostensibly refers to the code providing the currently active django
+    project.
+
+    Args:
+        model (django.db.models.Model): A model to check.
+
+    Returns:
+        bool: True if the given model is non-local.
+    """
+    # If the model _was_ local to the current IDA repository, it should be
+    # defined somewhere under sys.prefix + '/src/' or in a path that points to
+    # the current checked-out code.  On Posix systems according to our testing,
+    # non-local packages get installed to paths containing either
+    # "site-packages" or "dist-packages".
+    non_local_path_prefixes = []
+    for path in sys.path:
+        if 'dist-packages' in path or 'site-packages' in path:
+            non_local_path_prefixes.append(path)
+    model_source_path = inspect.getsourcefile(model)
+    return model_source_path.startswith(tuple(non_local_path_prefixes))
```

Comment:
@jmbowman or @nedbat can you think of a more robust way to check that a python module was installed from local source code vs. via pip/pypi (i.e. non-local)?  So far, the only difference between the two types of modules is that local ones tend to have one of these sourcefile prefixes:

```
/edx/app/edxapp/edx-platform/...
/edx/app/edxapp/venvs/edxapp/src/...
```
whereas non-local pip-installed ones have this:
```
/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/...
```
The current code takes advantage of the "site-packages" string being present only on non-local prefixes, but this low-level assumption may not be robust.

Reply:
The `/src/` component is because the package was installed with the -e flag.  That is not a guaranteed indicator that the code is "ours".  And we should be moving away from -e, so there is definitely be code we own that is being installed into site-packages.

Reply:
As we move away from -e, doesn't that also entail splitting that package out of edx-platform into its own repository?  That turns it into non-local as far as I'm concerned.

Reply:
I guess I misunderstood what you meant by "local" then.